pydrive2/fs/spec.py

```diff
@@ -49,13 +55,120 @@ def should_retry(exc):
     )(func)
 
 
+def _cache_file(client_id):
+    cache_dir = appdirs.user_cache_dir("pydrive2fs", appauthor=False)
+    os.makedirs(cache_dir, exist_ok=True)
+    return os.path.join(cache_dir, client_id + ".json")
+
+
 class GDriveFileSystem(AbstractFileSystem):
-    def __init__(self, path, google_auth, trash_only=True, **kwargs):
+    GDRIVE_CREDENTIALS_DATA = "GDRIVE_CREDENTIALS_DATA"
+
+    def __init__(
+        self,
+        path,
+        google_auth=None,
+        trash_only=True,
+        client_id=None,
+        client_secret=None,
+        client_user_email=None,
```

Comment:
it's strange that we can't make it part of kwargs tbh ... I would expect it to be also in that config?

Reply:
We can, it was a conscious choice to have them explicitly like that. But yeah, getting unwieldy now.